src/Runner.Worker/JobHookProvider.cs

```diff
@@ -60,7 +60,7 @@ public async Task RunHook(IExecutionContext executionContext, object data)
             Dictionary<string, string> inputs = new()
             {
                 ["path"] = hookData.Path,
-                ["shell"] = ScriptHandlerHelpers.GetDefaultShellForScript(hookData.Path, Trace, prependPath)
+                ["shell"] = ScriptHandlerHelpers.GetDefaultShellNameForScript(hookData.Path, Trace, prependPath)
```

Comment:
I think we may need to rethink how do do this in general, may require changes to script handler as well

Basically, we want to choose bash fallback to sh, or pwsh fallback to pwsh based on the file extension.

However, if we execute those shells, we want to use the default args for `sh` or `powershell`. The only big different here is we should **not** use bash's default args `bash --noprofile --norc -eo pipefail {0}` described [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell)

I think with your change, bash will be invoked using -eo pipefail, and we want to avoid that. Is there another way we can solve this problem? We may need to make changes to the script handler.

Reply:
I will dig into that. Thank you for the review!